2017-12-21 16:23:19 ERROR [com.zaxxer.hikari.pool.HikariPool.throwPoolInitializationException(HikariPool.java:566)] - HikariPool-1 - Exception during pool initialization.
com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:341) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2189) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2222) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2017) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:779) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:389) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:117) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:123) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:375) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:204) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:459) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:533) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:114) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:97) [HikariCP-2.7.4.jar:?]
	at com.cashier.dao.JdbcUtils.getConnection(JdbcUtils.java:55) [classes/:?]
	at com.cashier.service.impl.UserServiceImpl.isExist(UserServiceImpl.java:72) [classes/:?]
	at com.cashier.main.MainGUI.NewMembers(MainGUI.java:189) [classes/:?]
	at com.cashier.main.MainGUI$3.actionPerformed(MainGUI.java:132) [classes/:?]
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.DefaultButtonModel.setPressed(Unknown Source) [?:1.8.0_151]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at javax.swing.JComponent.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.processEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.processEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Window.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.access$500(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_151]
	at java.security.AccessController.doPrivileged(Native Method) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_151]
	at java.security.AccessController.doPrivileged(Native Method) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method) ~[?:1.8.0_151]
	at java.net.DualStackPlainSocketImpl.socketConnect(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.doConnect(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.PlainSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.SocksSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.Socket.connect(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:211) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:300) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	... 60 more
2017-12-21 16:23:21 ERROR [com.zaxxer.hikari.pool.HikariPool.throwPoolInitializationException(HikariPool.java:566)] - HikariPool-1 - Exception during pool initialization.
com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:341) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2189) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2222) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2017) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:779) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:389) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:117) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:123) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:375) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:204) ~[HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:459) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:533) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:114) [HikariCP-2.7.4.jar:?]
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:97) [HikariCP-2.7.4.jar:?]
	at com.cashier.dao.JdbcUtils.getConnection(JdbcUtils.java:55) [classes/:?]
	at com.cashier.service.impl.UserServiceImpl.register(UserServiceImpl.java:47) [classes/:?]
	at com.cashier.main.MainGUI.NewMembers(MainGUI.java:193) [classes/:?]
	at com.cashier.main.MainGUI$3.actionPerformed(MainGUI.java:132) [classes/:?]
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source) [?:1.8.0_151]
	at javax.swing.DefaultButtonModel.setPressed(Unknown Source) [?:1.8.0_151]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at javax.swing.JComponent.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.processEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.processEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Window.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.access$500(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_151]
	at java.security.AccessController.doPrivileged(Native Method) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_151]
	at java.security.AccessController.doPrivileged(Native Method) [?:1.8.0_151]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_151]
	at java.awt.EventQueue.dispatchEvent(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_151]
	at java.awt.EventDispatchThread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method) ~[?:1.8.0_151]
	at java.net.DualStackPlainSocketImpl.socketConnect(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.doConnect(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source) ~[?:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.PlainSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.SocksSocketImpl.connect(Unknown Source) ~[?:1.8.0_151]
	at java.net.Socket.connect(Unknown Source) ~[?:1.8.0_151]
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:211) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:300) ~[mysql-connector-java-5.1.44.jar:5.1.44]
	... 60 more
